File: decompiled/3.17.1/be/ehealth/businessconnector/genins/service/GenInsService.java
package be.ehealth.businessconnector.genins.service;

import be.ehealth.businessconnector.genins.exception.GenInsBusinessConnectorException;
import be.ehealth.technicalconnector.exception.SessionManagementException;
import be.ehealth.technicalconnector.exception.TechnicalConnectorException;
import be.ehealth.technicalconnector.service.sts.security.SAMLToken;
import be.fgov.ehealth.genericinsurability.protocol.v1.GetInsurabilityAsFlatResponse;
import be.fgov.ehealth.genericinsurability.protocol.v1.GetInsurabilityAsXmlOrFlatRequestType;
import be.fgov.ehealth.genericinsurability.protocol.v1.GetInsurabilityResponse;

public interface GenInsService {
   GetInsurabilityResponse getInsurability(SAMLToken var1, GetInsurabilityAsXmlOrFlatRequestType var2) throws GenInsBusinessConnectorException, TechnicalConnectorException, SessionManagementException;

   GetInsurabilityAsFlatResponse getInsurabilityAsFlat(SAMLToken var1, GetInsurabilityAsXmlOrFlatRequestType var2) throws GenInsBusinessConnectorException, TechnicalConnectorException, SessionManagementException;
}

File: src/main/kotlin/org/taktik/connector/business/genins/service/GenInsService.kt
/*
 *
 * Copyright (C) 2018 Taktik SA
 *
 * This file is part of FreeHealthConnector.
 *
 * FreeHealthConnector is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation.
 *
 * FreeHealthConnector is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with FreeHealthConnector.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package org.taktik.connector.business.genins.service

import org.taktik.connector.business.genins.exception.GenInsBusinessConnectorException
import org.taktik.connector.technical.exception.SessionManagementException
import org.taktik.connector.technical.exception.TechnicalConnectorException
import org.taktik.connector.technical.service.sts.security.SAMLToken
import be.fgov.ehealth.genericinsurability.protocol.v1.GetInsurabilityAsXmlOrFlatRequestType
import be.fgov.ehealth.genericinsurability.protocol.v1.GetInsurabilityResponse

interface GenInsService {
    @Throws(GenInsBusinessConnectorException::class, TechnicalConnectorException::class, SessionManagementException::class)
    fun getInsurability(token: SAMLToken, request: GetInsurabilityAsXmlOrFlatRequestType): GetInsurabilityResponse
}
